lib/index.js

```diff
@@ -47,78 +47,67 @@ module.exports = function loader (css, map) {
 
   validateOptions(require('./options.json'), options, 'PostCSS Loader')
 
-  const rc = {
-    path: path.dirname(file),
-    ctx: {
-      file: {
-        extname: path.extname(file),
-        dirname: path.dirname(file),
-        basename: path.basename(file)
-      },
-      options: {}
+  Promise.resolve().then(() => {
+    if (!options.config || (options.config && !options.config.path)) {
```

Comment:
@michael-ciniawsky current behavior not works as expected we should fix this, see https://github.com/postcss/postcss-loader/pull/234#issue-229052711 (point 2).